### DISCUSSION TITLE
Custom Docker Action can't find node script

### DISCUSSION THREAD

**Discussion by Muhlex** [community/community#25167] [Actions]
<p>I have been trying to setup a GitHub Action that runs a single node script (called index.js). This script has a node module as dependency, which is why I am not able to run it as a Javascript action and wanted to try using docker for the first time ever. Now GitHub Actions executes my dockerfile and sets up the container. Copying the package(-lock).json files and installing them via npm ci works perfectly. I then copy the script in the same directory and set the entrypoint accordingly.</p>
<pre><code>FROM node:lts

WORKDIR /github/workspace/

# Install app dependencies
COPY package*.json ./
RUN npm ci

# Bundle app source
COPY index.js ./
RUN ["chmod", "+x", "index.js"]

# Debug to verify that everything is setup correctly
RUN ls
RUN cd .. ; ls
RUN cd ../.. ; ls

ENTRYPOINT ["node", "index.js"]
</code></pre>
<p>When GitHub Actions then runs the container after building it, it automatically sets the working directory to /github/workspace, which is why I have replicated this in my Dockerfile ( WORKDIR /github/workspace/ ).</p>
<p>Using the unelegant ls outputs (forgive me, I’m new to Linux scripting in general), I can verify that the script and node_modules folder are located in /github/workspace/ .</p>
<p>When the container is then run, it instantly terminates with the following error:</p>
<pre><code>internal/modules/cjs/loader.js:985
  throw err;
  ^

Error: Cannot find module '/github/workspace/index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:982:15)
    at Function.Module._load (internal/modules/cjs/loader.js:864:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
</code></pre>
<p>This suggests that the file does not exist. The debug output I added when building the container shows me otherwise though (root directory shortened for legibility):</p>
<pre><code>Step 7/10 : RUN ls
 ---&gt; Running in ef7c06d4e9f2
index.js
node_modules
package-lock.json
package.json
Removing intermediate container ef7c06d4e9f2
 ---&gt; 155562cd55ca
Step 8/10 : RUN cd .. ; ls
 ---&gt; Running in 1769785ff14c
workspace
Removing intermediate container 1769785ff14c
 ---&gt; 71d08ac06ceb
Step 9/10 : RUN cd ../.. ; ls
 ---&gt; Running in 470c23db0ef6
...
github
...
Removing intermediate container 470c23db0ef6
 ---&gt; 48524c4c0c9e
</code></pre>
<p>Am I missing something obvious here? When I build the docker container locally and then execute it, it works perfectly fine.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/muhlex">@muhlex</a> , </p>
<p>Please don’t set the working directory to  /github/workspace in the action Dockerfile. You could just copy it to the root path. Please try to the method below: </p>
<pre><code>FROM node:lts

# Bundle app source
COPY index.js /index.js
RUN ["chmod", "+x", "index.js"]

# Debug to verify that everything is setup correctly
RUN ls
ENTRYPOINT ["node", "/index.js"]
</code></pre>
<p> </p>

**Comment by Muhlex** [comment]
<p>Thanks a lot. That actually does it.</p>
<p>I tried doing it that way from the beginning but probably messed up something in regards to relative and absolute file paths.</p>
